### ISSUE TITLE
Feature: Notifications

### ISSUE THREAD

**Issue by famfop** [LemonBoy/ldm#73]
Hi,
I believe I once asked but maybe I am mistaking. So I was wondering if there is a way to get (custom?) notifications on mount / unmount? I am using the systemd service. Even if it is possible, I'd like to see a / some simple examples on how to configure it.

**Comment by LemonBoy** [comment]
Please check out the manual, the `-c` option allows you to specify a custom script that `ldm` invokes every time a device is mounted/unmounted. The `LDM_ACTION`, `LDM_FS`, `LDM_NODE` and `LDM_MOUNTPOINT` environment variables are defined and should be enough to generate the notification blurb.

The only problem I see is that you might need to export the `DISPLAY` variable for `notify-send` to work from within the script.

**Comment by famfop** [comment]
Thanks, so do I need to export it inside the script or how would you do that?

**Comment by LemonBoy** [comment]
I guess something like
```shell
DISPLAY=:1 notify-send "Hello"
```
is enough, just make sure to export the correct display.

**Comment by famfop** [comment]
I have been trying different things and cannot get it going. I tried it the way you said, also setting `DISPLAY=:0.0`, `DISPLAY=:0 and some other things but it didn't work. Do you have any idea?

**Comment by LemonBoy** [comment]
Not really, make sure the script is executable and is actually executed first.

**Comment by famfop** [comment]
I was on holiday and am back now. Let me summarize what I tried (unsuccesfully):
I made a scriptfile and made it executable. My `$DISPLAY` variable said `:0.0` so I was not sure and tried out `DISPLAY=:0 notify-send "test"`,  `DISPLAY=:0.0 notify-send "test"`, for I don't know what reason `DISPLAY=:1 notify-send "test"` and also "plain" `notify-send "test"`.
I edited my systemd file (the `ExecStart` variable) such that it is `ExecStart=/usr/bin/ldm -u ${MOUNT_OWNER} -p ${BASE_MOUNTPOINT} -c /path/to/executable/scriptfile.sh

I am still not able to get the output "test" from my notification daemon... Do you see any mistakes / are you able to get notifications?

**Comment by LemonBoy** [comment]
There's no output at all, the file descriptors for `stdin`, `stdout` and `stderr` are closed before `sh` is invoked. Please check your syslog (`journalctl` or whatever) for `ldm`-related errors and try something like this to see if the commands are executed.

```sh
echo "Hello" > /tmp/check
```

**Comment by famfop** [comment]
Ok this worked, no error and the file was produced. But the notify-send just does not work. I went on reading and stumbled upon some posts seemingly having the same issue with udev rules. The problem seems to be the at startup missing `DBUS_SESSION_BUS_ADDRESS`. I read a bit and tried but I don't have too much time right know, so I just leave the - in my opinion - most helpful link [here](https://hackaday.com/2009/09/18/how-to-write-udev-rules/). Especially look at the Resources, mainly: [notify-send bug](https://bugs.launchpad.net/ubuntu/+source/libnotify/+bug/160598) and [notify-send workaround](http://ubuntuforums.org/showthread.php?p=6889270#post6889270).
